kernel/src/snapshot.rs

```diff
@@ -413,6 +442,30 @@ impl Snapshot {
             None => Err(Error::generic("Last commit file not found in log segment")),
         }
     }
+
+    fn build_table_configuration(
```

Comment:
This is basically a Snapshot constructor. Let's make the name clearer.
```suggestion
    fn try_new_from_log_segment_impl(
```